models/DataObject/Service.php

```diff
@@ -724,6 +705,62 @@ private static function getValueForObject($object, $key, $brickType = null, $bri
         return $result;
     }
 
+    /**
+     * gets store value for given object and key
+     *
+     * @static
+     *
+     * @param Concrete $object
+     * @param string $key
+     * @param string|null $requestedLanguage
+     *
+     * @return string
+     */
+    private static function getStoreValueForObject($object, $key, $requestedLanguage)
+    {
+        $keyParts = explode('~', $key);
+
+        if (strpos($key, '~') === 0) {
+            $type = $keyParts[1];
+            if ($type === 'classificationstore') {
+                $field = $keyParts[2];
+                $groupKeyId = explode('-', $keyParts[3]);
+
+                $groupId = $groupKeyId[0];
+                $keyid = $groupKeyId[1];
+                $getter = 'get' . ucfirst($field);
+
+                if (method_exists($object, $getter)) {
+                    /** @var Classificationstore $classificationStoreData */
+                    $classificationStoreData = $object->$getter();
+
+                    /** @var Model\DataObject\ClassDefinition\Data\Classificationstore $csFieldDefinition */
+                    $csFieldDefinition = $object->getClass()->getFieldDefinition($field);
+                    $csLanguage = $requestedLanguage;
+
+                    if (!$csFieldDefinition->isLocalized()) {
+                        $csLanguage = 'default';
+                    }
+
+                    $fielddata = $classificationStoreData->getLocalizedKeyValue($groupId, $keyid, $csLanguage, true, true);
+
+                    $keyConfig = Model\DataObject\Classificationstore\KeyConfig::getById($keyid);
+                    $type = $keyConfig->getType();
+                    $definition = json_decode($keyConfig->getDefinition());
+                    $definition = \Pimcore\Model\DataObject\Classificationstore\Service::getFieldDefinitionFromJson($definition, $type);
+
+                    if (method_exists($definition, 'getDataForGrid')) {
+                        $fielddata = $definition->getDataForGrid($fielddata, $object);
+                    }
+
+                    return $fielddata;
+                }
+            }
+        }
+
+        return '';
```

Comment:
returning an empty string here is changing the original behavior

Reply:
Okay in order to not change the original behavior, I return "null" like it was before.